proto/src/determined/rbac/v1/rbac.proto

```diff
@@ -19,10 +19,74 @@ message Role {
 // Permission represents an action a user can take in the system
 message Permission {
   // The id of the permission
-  int32 id = 1;
+  PermissionType id = 1;
   // The name of the permission
   string name = 2;
-  // Whether the permission is a globl-only permission
+  // Whether the permission is a global-only permission
+  bool is_global = 3;
+}
+
+// List of permissions types.
+enum PermissionType {
+  // The permission type is unknown.
+  PERMISSION_TYPE_UNSPECIFIED = 0;
+
+  // Can create and update other users.
+  // Allows updating other users passwords making this
+  // permission give all other permissions effectively.
+  PERMISSION_TYPE_ADMINISTRATE_USER = 91001;
+
+  // Ability to create experiments.
+  PERMISSION_TYPE_CREATE_EXPERIMENT = 2001;
+  // Ability to view experiment's model code, checkpoints, trials.
+  PERMISSION_TYPE_VIEW_EXPERIMENT_ARTIFACTS = 2002;
+  // Ability to view experiment's metadata like experiment config, progress.
+  PERMISSION_TYPE_VIEW_EXPERIMENT_METADATA = 2003;
+  // Ability to update experiment and experiment's lifecycle.
+  PERMISSION_TYPE_UPDATE_EXPERIMENT = 2004;
+  // Ability to update experiment's metadata.
+  PERMISSION_TYPE_UPDATE_EXPERIMENT_METADATA = 2005;
+  // Ability to delete experiment.
+  PERMISSION_TYPE_DELETE_EXPERIMENT = 2006;
+
+  // Ability to create, update, and add / remove users from groups.
+  PERMISSION_TYPE_UPDATE_GROUP = 93001;
+
+  // Ability to create workspaces.
+  PERMISSION_TYPE_CREATE_WORKSPACE = 94001;
+  // Ability to view workspace.
+  PERMISSION_TYPE_VIEW_WORKSPACE = 4002;
+  // Ability to update workspace.
+  PERMISSION_TYPE_UPDATE_WORKSPACE = 4003;
+  // Ability to delete workspace.
+  PERMISSION_TYPE_DELETE_WORKSPACE = 4004;
+
+  // Ability to create projects.
+  PERMISSION_TYPE_CREATE_PROJECT = 5001;
```

Comment:
or should this be 95001? although I am not sure what's the case for that.

Reply:
9xxxx refers to it being global scope only permission. Though I'll add a comment explaining it.